aten/src/THC/generic/THCTensorIndex.cu

```diff
@@ -145,14 +183,24 @@ void THCTensor_(indexCopy)(THCState *state, THCTensor *dst, int dim, THCudaLongT
         SMALL_INDEX(real, unsigned int, -1, -1, -1);
       }
     } else {
+      bool indexIsMajor = THCTensor_(indexShouldBeMajor)(dstInfo, dstCopyDim);
+
       if (dstInfo.dims == 1 && srcInfo.dims == 1 && indContig) {
-        LARGE_INDEX(real, unsigned int, 1, 1, -2);
+        LARGE_INDEX(real, unsigned int, 1, 1, -2, true);
       } else if (dstInfo.dims == 2 && srcInfo.dims == 2 && indContig) {
-        LARGE_INDEX(real, unsigned int, 2, 2, -2);
+        if (indexIsMajor) {
+          LARGE_INDEX(real, unsigned int, 2, 2, -2, true);
+        } else {
+          LARGE_INDEX(real, unsigned int, 2, 2, -2, false);
+        }
```

Comment:
Why can't `indexIsMajor` just be a macro argument instead of a literal?

Reply:
It's used as a template argument (so that the `if` conditionals I added to CUDA kernels are determined at compile time).